src/main/java/seedu/address/model/contact/Name.java

```diff
@@ -3,20 +3,29 @@
 import static java.util.Objects.requireNonNull;
 import static seedu.address.commons.util.AppUtil.checkArgument;
 
+import java.util.List;
+
 /**
  * Represents a Contact's name in the address book.
  * Guarantees: immutable; is valid as declared in {@link #isValidName(String)}
  */
 public class Name {
+    public static final String ALIAS = "@";
+    public static final String SON_OF = "S/O";
+    public static final String DAUGHTER_OF = "D/O";
+    public static final String BRACKET_PAIR = "(<Name>)";
 
     public static final String MESSAGE_CONSTRAINTS =
-            "Names should only contain alphanumeric characters and spaces, and it should not be blank";
-
+            "Names should only contain letter characters and spaces, it should neither be blank nor "
+                    + "contain numbers, with the following exceptions: " + ALIAS + ", " + SON_OF + ", "
+                    + DAUGHTER_OF + ", " + BRACKET_PAIR + ". These exceptions can be used, but must only appear once."
+                    + ALIAS + ", " + SON_OF + ", " + DAUGHTER_OF + " must be in the middle of the name while "
```

Comment:
+ ALIAS + " OR " + SON_OF + " OR " + DAUGHTER_OF + " must be in the middle of the name while "